spec/features/user_oauth2_spec.rb

```diff
@@ -0,0 +1,106 @@
+require 'rails_helper'
+RSpec.describe 'User oauth2 registration and sign in', type: :feature do
+  before do
+    OmniAuth.config.test_mode = true
+  end
+
+  shared_examples_for :an_oauth2_provider do
```

Comment:
I like this use of `shared_examples_for`, it makes this test more DRY and allows for more efficient/clean addition/removal of providers.